### PR TITLE
Add more config settings

### PR DESCRIPTION
**Pull request by imorland**
Focussing directly on the scheduler. Adding 4 config settings:

Screenshot 2020-02-06 at 21 11 01

1. Added option to remind to only discussions in certain tags (if flarum/tags is enabled)
2. For infrastructure using memcache or redis, allows the option to use the `onOneServer()` directive. Default `off`. [Info](https://laravel.com/docs/5.8/scheduling#running-tasks-on-one-server)
3. Option to set the best answer reminder task to not run during unsocial hours. Default `off` [Info](https://laravel.com/docs/5.8/scheduling#scheduling-shell-commands)
4. Current implementation is to write to `logs/fof-best-answer.log`. Option to disable writing of the log file, if required. Default `on`

@datitisev I'm extremely open to ideas and suggestions here. There may well be a better way to handle this situation.

### CODE REVIEW

**Comment by imorland** [comment]
Using `Carbon::now()`, rather than `Carbon::today()`, we spread the sending of notifications over the day.

Added option to remind to only discussions in certain tags (if flarum/tags is enabled)

Excluded private discussions from byobu (if enabled)

**Comment by datitisev** [comment]
I wrote a response that I apparently did not send - whoops.

For the scheduler changes, as far as I can tell, these modify the settings for ALL jobs processed by the queue - not just this extension's. Which makes me think those changes would be more appropiate in the driver's package (e.g. bokt redis if that's what you are using). 

So with the log file there, that just looks to me like it'll log EVERY job, not just this extension's, into the fof best answer log file.

Am I wrong?

---

Also - I'd suggest removing the check for if byobu enabled as other extensions can take advantage of `$discussion->is_private`, it's not byobu-specific.

**Comment by imorland** [comment]
I can confirm that the scheduler changes do indeed only modify this extension's jobs. This was tested alongside some other (currently private :( ) giffgaff extensions which make extensive use of the scheduler. This includes the logfile - is there a particular reason this is currently enabled permanently at the moment?

Good point on the `is_private` column, I wrongly assumed it was a migration supplied by byobu. Now I've double checked byobu's migrations, I see it's not specific. Have removed the check.

File: src/Console/NotifyCommand.php [hunk 1]
```diff
@@ -59,21 +66,32 @@ public function __construct(SettingsRepositoryInterface $settings, NotificationS
     public function handle()
     {
         $days = (int) $this->settings->get('fof-best-answer.select_best_answer_reminder_days');
-        $time = Carbon::today()->addDay(-$days);
+        $time = Carbon::now()->addDay(-$days);
```
Comment (luceos) [inline src/Console/NotifyCommand.php]: Should use the subDays method with a positive argument.

Comment (imorland) [reply src/Console/NotifyCommand.php]: Change implemented :)

**Comment by imorland** [comment]
@luceos I've attempted this change to the default settings assignment, could potentially be improved some more in future, but this does work..

**Comment by imorland** [comment]
I've removed the [WIP] tag from this PR, as I feel it's getting close to completion, subject to review :)

**Comment by datitisev** [comment]
Ah I apologize, I skipped the part where it sets the command to the Notify command.

We may want to move these settings to a sort-of "Advanced" section and/or add help text that explains what it does and why/if the forum may need it enabled/disabled.

**Comment by datitisev** [comment]
Are there going to be any more PRs from giffgaff or are all these features ready for release?

**Comment by imorland** [comment]
Thanks guys.

@datitisev - there will be at least one more. We would like to add a toggle to have the ‘set/unset’ best answer either where it currently is, or in it’s previous position in the post footer.

**Comment by imorland** [comment]
That said, feel free to release the features so far, if you wish 🙂

**Pull request merged**